index.js

```diff
@@ -245,15 +245,26 @@ ShareDbMongo.prototype.commit = function(collectionName, id, op, snapshot, optio
     self._writeSnapshot(request, id, snapshot, opId, function(err, succeeded) {
       if (succeeded) return callback(err, succeeded);
       // Cleanup unsuccessful op if snapshot write failed. This is not
-      // neccessary for data correctness, but it gets rid of clutter
+      // necessary for data correctness, but it gets rid of clutter
       self._deleteOp(request.collectionName, opId, function(removeErr) {
         callback(err || removeErr, succeeded);
       });
     });
   });
 };
 
-function createRequestForMiddleware(options, collectionName, op) {
+function createRequestForMiddleware(options, collectionName, op, fields) {
+  // When we're creating a request for submitting an op, let downstream middleware know.
+  if (fields && fields.$submit === true) {
+    if (!options) {
+      options = {};
+    }
+    /**
+     * TODO What if sharedb populated this?
```

Comment:
(On this, in future I think we should potentially provide the corresponding ShareDB [lifecycle hook](https://github.com/share/sharedb/blob/5c9148dc44d2c43582386875ed76c08070cfce3f/lib/backend.js#L49-L62), but definitely out-of-scope for this change)